src/components/pages/SearchResultsPage.js

```diff
@@ -1,88 +1,264 @@
+    { key: "yearLow", title: "Year Low" },
+  ];
+
+  // Aggregated object of the stock data to pass as a prop to children components
+  // using the state variables from above
+  const aggregatedStockData = [
     {
-      symbol: "AAPL",
-      recentPrice: 150.23,
-      open: 140.0,
-      high: 162.5,
-      low: 126.5,
-      yearHigh: 167.0,
-      yearLow: 100.0,
-      percentChanged: 1.5,
+      recentPrice: recentPrice,
+      open: open,
+      high: high,
+      low: low,
+      yearHigh: high52w,
+      yearLow: low52w,
+      priceChange: priceChange,
+      percentChanged: percentChanged,
+      marketClosed: marketClosed,
+      dateClosed: dateClosed,
     },
-  ]);
+  ];
 
   /*
-  Callback used to get more data based on the filter. Each time any of the buttons 
-  are clicked this will be called to get more data. This will update the chart data which 
-  will then re-render the graph
+ Function to determine what timeframe of graph data to fetch based on the filter enum (timeframe selected by user)
+  calls the getGraphData to retreive data and updates the filter the user has selcted with: setSelectedFilter
   */
   const getData = (filter) => {
-    if (filter === "Past Year") {
-      setChartData([]);
-    } else {
-      setChartData(chartData.concat(chartData));
-      setCategories(categories.concat(categories));
-      setStockData(stockData);
-      console.log("Request filter: " + filter);
+    switch (filter) {
+      case filters.DAY:
+        getGraphData("D");
+        setSelectedFilter(filter);
+        break;
+      case filters.FIVE:
+        getGraphData("5D");
+        setSelectedFilter(filter);
+        break;
+      case filters.MONTH:
+        getGraphData("M");
+        setSelectedFilter(filter);
+        break;
+      case filters.YEAR:
+        getGraphData("Y");
+        setSelectedFilter(filter);
+        break;
+      default:
+        break;
     }
   };
 
-  // Should initial get all the data for the day time frame
-  //useEffect(() => {
-  //  `getData("day")
-  //})
+  // Callback function that calls Algoquant's api to grab new graph data based on the timeframe
+  const getGraphData = useCallback(
+    (timeframe) => {
+      if (algoquantApi.token) {
+        setGraphLoading(true);
+        algoquantApi
+          .getGraphData(location.state.value, timeframe)
+          .then((resp) => {
+            setXValues(resp.data["close"]);
+            setPercentChanged(resp.data["percent_change"].toFixed(2));
+            setPriceChange(
+              parseFloat(resp.data["interval_price_change"]).toFixed(2)
+            );
+            setMarketClosed(resp.data["is_market_closed"]);
+
+            // If the timeframe selected was day, store the first timeframe (yVal) to keep track of the day the market was open,
+            // DateClosed variable will then used to show the date the market is closed, if it is.
+            if (timeframe === "D") {
+              setDateClosed(
```

Comment:
Couldn't this just be moved to the switch statement below for `case "D"` ?

Reply:
fixed